kaldb/src/main/java/com/slack/kaldb/server/RecoveryTaskCreator.java

```diff
@@ -180,7 +182,7 @@ public long determineStartingOffset(long currentHeadOffsetForPartition) {
     List<RecoveryTaskMetadata> recoveryTasks = recoveryTaskMetadataStore.listSync();
     long highestDurableOffsetForPartition =
         getHighestDurableOffsetForPartition(
-            nonLiveSnapshotsForPartition, recoveryTasks, partitionId);
+            beginningOffSetForPartition, nonLiveSnapshotsForPartition, recoveryTasks, partitionId);
```

Comment:
the idea here is with this change `highestDurableOffsetForPartition` will return `beginningOffSetForPartition` when there is no metadata.